### ISSUE TITLE
Bad video quality!?

### ISSUE THREAD

**Issue by hmaalen** [rFlex/SCRecorder#380]
The example project and my own usage of the lib results in quite bad quality video (pixelation etc).
A bit surprised this has not been mentioned by anyone else. Am I the only one having this problem?

**Comment by rFlex** [comment]
Which device do you use? The quality should be configurable through the bitrate property of videoConfiguration.

**Comment by howeguo** [comment]
hi,@rFlex , I found the issue of the bad  quality in iPhone7, can you tell me what is the exact bitrate in iPhone 7 ? thanks very much

**Comment by ruslanmoskalenko** [comment]
I'm seeing the same problem. I use AVAssetExportPresetHighestQuality but quality video is bad(pixelated). My device iPhone 6s. Have suggestions how to fix it?

**Comment by dyadenkor** [comment]
@rFlex Problem exist only on iPhone 7. Seems there problem with hardware. Is it possible for you to check?

**Comment by jjsandgreen** [comment]
@rFlex are you alive? :)

**Comment by wojczitsu** [comment]
@rFlex same here

**Comment by freeubi** [comment]
just set to a better resolution

**Comment by FatihGirisken** [comment]
The same here @rFlex 
I tried every resolution and every setting, the video is pixelated each time. Please, help!

**Comment by MMasterson** [comment]
Have you guys looked at the PRs or forks? Seems this fork is the most active and advanced.

https://github.com/lucaji/SCRecorder

**Comment by FatihGirisken** [comment]
yeah, I have checked them. I dont know how no one noticed this problem. Its mostly pixelated, and also depending on the light in the room, it gets better or worse. Is it some exposure setting or sth maybe? Please, open the native camera, or to any other video app, and compare the video quality. I am thinking to change the library because of it.

**Comment by freeubi** [comment]
I dont have any problem like this. Go and check your exported videos quality, how is it exported (not the setti gs, the real video file)

**Comment by FatihGirisken** [comment]
Exported video is the same. I use front camera, iPhone7, AVCaptureSessionPresetHigh. And similar settings in the export config. SCPresetHighestQuality, SCContextTypeAuto, AVFileTypeMPEG4.

**Comment by FatihGirisken** [comment]
@freeubi are you using a different fork? and which device?
Pls, see this screenshot from a video file exported. This is obviously not the highest quality video on an iPhone-7.
![img_4122](https://user-images.githubusercontent.com/3841240/36076840-32ef0f96-0f62-11e8-91a7-bc558a358568.PNG)

**Comment by freeubi** [comment]
This is my settings: 
  ```
assetExportSession.outputFileType = AVFileTypeQuickTimeMovie;
                    
                    assetExportSession.audioConfiguration.preset = SCPresetMediumQuality;
                    
                    assetExportSession.videoConfiguration.bitrate = 250000;
                    assetExportSession.videoConfiguration.maxFrameRate = 24;
                    assetExportSession.videoConfiguration.codec = AVVideoCodecH264;
                    assetExportSession.videoConfiguration.size = CGSizeMake(360, 640);
```

i dont know wihtch version, because i dont use pods.
From the image i see low bitrate, try check the exported file data.

**Comment by dimohamdy** [comment]
you can fix it by change default bitrate value "kSCVideoConfigurationDefaultBitrate" to very large number

#define kSCVideoConfigurationDefaultBitrate 20000000000000

**Comment by SerjPridestar** [comment]
In my case I had the same problem however, I had a configuration that was responsible for this:

`exportSession.videoConfiguration.shouldKeepOnlyKeyFrames = true`

When I removed the following the video was agreeably much better.

Also, what might also cause an issue is the `maxFrameRate`.
